src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -44,6 +45,31 @@ public void Inject(SpanContext context, IHeadersCollection headers)
                 samplingPriority?.ToString(InvariantCulture));
         }
 
+        /// <summary>
+        /// Propagates the specified context by adding new  headers to a <see cref="HttpRequestHeaders"/>.
+        /// This locks the sampling priority for <paramref name="context"/>.
+        /// </summary>
+        /// <param name="context">A <see cref="SpanContext"/> value that will be propagated into <paramref name="headers"/>.</param>
+        /// <param name="headers">A <see cref="HttpRequestHeaders"/> to add new headers to.</param>
+        public void Inject(SpanContext context, HttpRequestHeaders headers)
```

Comment:
This class was designed to extract and inject headers into `IHeadersCollection` only and not take dependency on all the different header collection types that exist in dotnet. To use it on `HttpRequestHeaders`, consider wrapping that in a [`Datadog.Trace.Headers.HttpHeadersCollection`](https://github.com/DataDog/dd-trace-dotnet/blob/80b4c4dbb034e52a656941ca1ff7678d76a8daf7/src/Datadog.Trace/Headers/HttpHeadersCollection.cs). We have a [`HttpHeaders.Wrap()`](https://github.com/DataDog/dd-trace-dotnet/blob/80b4c4dbb034e52a656941ca1ff7678d76a8daf7/src/Datadog.Trace/ExtensionMethods/HttpHeadersExtensions.cs#L17) extension method to make this a little easier.

Reply:
See [`src/Datadog.Trace/Headers/`](https://github.com/DataDog/dd-trace-dotnet/tree/80b4c4dbb034e52a656941ca1ff7678d76a8daf7/src/Datadog.Trace/Headers) for implementations of `IHeadersCollection`.

Reply:
I've replaced the `HttpClientIntegration` with `HttpMessageInvokerIntegration` which allows correctly adding the distributed-tracing headers. It also does not require the use of the LoaderOptimization.

Reply:
I think this method is no longer used so it can be removed.